### PR TITLE
Use pyproject in prod container

### PR DESCRIPTION
**Pull request by GernotMaier**
Please review after #40 and #41.

No need to use is conda env create in the production container - we can do now 'pip install'.

Note that conda is still needed for a consistent pip / python installation (if you have a lot of time...search the web and try do to it without conda...).

### CODE REVIEW

**Comment by orelgueta** [comment]
I removed the use of conda now and saved us 120MB in the image size and about 20 seconds in build time (I know you like reducing the size of the container). Since we both made changes, perhaps @VictorBarbosaMartins should review this PR (once tests pass)?

**Comment by GernotMaier** [comment]
Agree (on Victor reviewing it)

**Comment by orelgueta** [comment]
I also piggybacked on this PR to remove the gzip extraction method when extracting the corsika tarball since it causes a failure with the newer tarball.

**Comment by GernotMaier** [comment]
I would suggest to ignore the failing test on build-simtools-container (dev):

- this PR is about the prod container
- open PR #45 has extensive updates to the dev container

**Comment by VictorBarbosaMartins** [comment]
Ok, I am having a look at it, and trying to get used to the procedure of building and running the container. I am commenting here and not in the review because my comments not necessarily are change requests.

**Comment by VictorBarbosaMartins** [comment]
The first comment is about a conda directory. I have built and run the docker after checking out the branch. It was really fast but I still see a conda directory there. as Gernot commented, we still need it to properly install pip. So assume this is ok? The directory alone has almost 1 Gb.
```
(base) root@3567dc85d35c:/workdir# cd conda
(base) root@3567dc85d35c:/workdir/conda# ls
LICENSE.txt  compiler_compat  condabin  etc      lib  pkgs  share  ssl                     x86_64-conda_cos7-linux-gnu
bin          conda-meta       envs      include  man  sbin  shell  x86_64-conda-linux-gnu
(base) root@3567dc85d35c:/workdir/conda# du -sh
981M	.
```

**Comment by VictorBarbosaMartins** [comment]
Probably unimportant, but I also found a `environment.yml.tmp` file inside `/workdir/simtools`. Not sure how it was produced but I think we dont need that, right? Should we just ignore it?

**Review by VictorBarbosaMartins — APPROVED** [review]
Building and running the container worked. testing the production application worked. despite my comments, I am approving and leaving to you to decide what to do with the comments. 
Thanks for improving the container!

**Comment by orelgueta** [comment]
I cannot see neither a conda directory nor an `environment.yml.tmp` file. Could it be that you ran a different image than the one you built?
I will wait to figure this out before merging, just in case.

**Comment by VictorBarbosaMartins** [comment]
> I cannot see neither a conda directory nor an `environment.yml.tmp` file. Could it be that you ran a different image than the one you built? I will wait to figure this out before merging, just in case.

That is strange, because I started from scratch (deleted everything) , and followed the instructions here: https://github.com/gammasim/containers/tree/main/prod
I previously deleted all the containers I had before. We can check together tomorrow if you like.

**Comment by orelgueta** [comment]
But after building the container did you use the following command from the README to run?

`docker run --rm -it -v "$(pwd)/external:/workdir/external" ghcr.io/gammasim/simtools-prod:latest bash`

**Comment by VictorBarbosaMartins** [comment]
> But after building the container did you use the following command from the README to run?
> 
> `docker run --rm -it -v "$(pwd)/external:/workdir/external" ghcr.io/gammasim/simtools-prod:latest bash`

yes, I hope that is what we are supposed to do?

**Comment by orelgueta** [comment]
No, this would download the image which is built by GitHub and run that one. It would be for the current main branch since it's built on PR (or on release? not sure). You need to run the container you just built by replacing `ghcr.io/gammasim/simtools-prod:latest` with the name you gave it. If you used the default it would be 'simtools-prod`. 

Maybe this README should be updated with these instructions after the "Building" section.

**Comment by VictorBarbosaMartins** [comment]
> No, this would download the image which is built by GitHub and run that one. It would be for the current main branch since it's built on PR (or on release? not sure). You need to run the container you just built by replacing `ghcr.io/gammasim/simtools-prod:latest` with the name you gave it. If you used the default it would be 'simtools-prod`.
> 
> Maybe this README should be updated with these instructions after the "Building" section.

ok, thanks for clarifying. now I understand the reason why "building" and "running" are shown separated in the instructions :) 
I agree, we need more precise instructions.

**Comment by orelgueta** [comment]
OK so does that mean we can merge this? Did you check the container you built?

BTW, the instructions for building and running are not entirely clear here because this prod container is not meant to be built by a user but essentially only automatically. The only other case is a developer testing it like we do now. We could add instructions for developers  though if you think it's necessary.

**Comment by VictorBarbosaMartins** [comment]
Hi Orel. Yes, please add some further instructions for completeness and then lets merge it :)

**Comment by orelgueta** [comment]
OK I added instructions and also added a command to the Dockerfile to remove the script we downloaded to install pip.

@GernotMaier, why in the instructions for running the prod container we also setup an `external` directory? I think in this case it is not actually necessary, no?

File: prod/README.md
Comment (GernotMaier) [inline prod/README.md]: What does this sentence mean? We don't use it because we don't set it? Maybe better to say that `In case file exchange with the local file system is required, use the docker syntax to mount a directory (e.g., `docker run --rm -it -v "$(pwd)/external:/workdir/external" ghcr.io/gammasim/simtools-prod:latest ...`).

Comment (GernotMaier) [reply prod/README.md]: Also think that this section **Running** should be completely removed. There is a running section further up in the readme. I'll make a suggestion.

Comment (orelgueta) [reply prod/README.md]: Your suggestion is better for setting up an external directory.

For the running section, I agree it exists but it confused Victor cause it runs the docker image from github rather than the one just built. I will wait for your concrete suggestion before commenting though.

File: prod/README.md
Comment (orelgueta) [inline prod/README.md]: This sentence should be in singular I think cause there's only one example below.

**Comment by orelgueta** [comment]
Your changes look fine to me, I think you can merge.

**Pull request merged**
